code/script.js

```diff
@@ -221,123 +268,244 @@ const CHARACTERS = [
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
-    }
+      alert(
+        `No, the person doesn't wear ${questions.options[questions.selectedIndex].value}! Remove all that wears ${group}`
+      )
+    } 
+  } else if (group === 'smoker') {
+      if (keep) {
+      alert(
+        `Yes, the person is a ${questions.options[questions.selectedIndex].value}! Keep all ${group}`
+      )
+    } else {
+    alert(
+      `No, the person is not a ${questions.options[questions.selectedIndex].value}! Remove all ${group}`
+      )
+    } 
   }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  else if (group === 'hairColor') {
+    if (keep) { 
+      alert(
+        `Yes, the person has ${questions.options[questions.selectedIndex].value} hair! Keep all persons with ${attribute} hair`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have ${questions.options[questions.selectedIndex].value} hair! Remove all persons with ${attribute} hair`
+      )
+    }
+  } 
+  else if (group === 'eyeColor') {
+    if (keep) { 
+      alert(
+        `Yes, the person has eyecolor ${questions.options[questions.selectedIndex].value}! Keep all persons with eyecolor ${attribute}`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have eyecolor ${questions.options[questions.selectedIndex].value}! Remove all persons with eyecolor ${attribute} `
+      )
+    } 
+  } 
+  else if (group === 'earrings' || group === 'beard') {
+    if (keep) { 
+      alert(
+        `Yes, the person wears ${questions.options[questions.selectedIndex].value}! Keep all persons with ${group}`
+        )
+    } else {
+      alert(
+        `No, the person doesn't wear ${questions.options[questions.selectedIndex].value}! Remove all persons with ${group} `
+      )
+    } 
+  } 
+  generateBoard(); 
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirm('Are you sure you want to guess?')) {
+    console.log('User want to make a guess!');
+    checkMyGuess(suspect)
+  } else {
+    console.log('The game continues!');
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  console.log(suspect);
```

Comment:
I learned previously (from earlier project) that i am inconsistent with my semicolons - i use them a little bit here and a little bit here. What i did for my project this time was to really go through the code to remove the ones that i added. But i am sure i still got one or two still being there :P

Reply:
Semicolons should be chacked and automatically fixed by VScode. Go to the settings and configure your prettier to do so.

Reply:
Thanks for the comments both of you, I have checked my settings in VS code.